tests/autoyast/wicked.pm

```diff
@@ -41,8 +41,12 @@ sub run {
     type_string "systemctl status wickedd.service\n";
     type_string "echo `wicked show all |cut -d ' ' -f 1` END | tee /dev/$serialdev\n";
     my $iflist = wait_serial("END", 10);
-    $iflist =~ s/\bEND\b//g;
-    $iflist =~ s/\blo\b//g;
+    # For poo#70453, to filter network link from mixed info of the output of wicked cmd
+    # we need substr output string from serial. Ex: 'lo0 eth0 END' -> 'eth0'
+    my $p1 = index($iflist, 'lo');
+    my $p2 = index($iflist, 'END');
+    my $str = substr($iflist, $p1+3, $p2-$p1-4);
+    $iflist = $str;
```

Comment:
i believe that you can this in one liner
```
($iflist) =~ s/(eth0)/$1/;
```
This should match `eth0` on `lo eth0 END` and assign it back to the variable.

Reply:
Thanks, I tried the line, but it is not work for me and just return the whole original $iflist.
In fact, if I can get the line of 'lo0 eth0 END' from the $iflist, I won't need to use substr to do it.

my $str = "12381819273918723
           ajsdfkljaksdfjfkasdf
           lo eth0 END";
print "str=$str\n";
($str) =~ s/(eth0)/$1/;
print "got str=$str\n";
lemon@linux-hxjt:~/perl> ./tt01.pl
str=12381819273918723
           ajsdfkljaksdfjfkasdf
           lo eth0 END
got str=12381819273918723
           ajsdfkljaksdfjfkasdf
           lo eth0 END

Reply:
It should be a list which need match between lo and END, such as "eth0 eth1 ethx"，so i suggest using following:

($iflist) =~ s/(lo\s)(.*)(\sEND)/$2/;

Reply:
The last line got 'eth0', but still return with previous noise info.

str=12381819273918723
           ajsdfkljaksdfjfkasdf
           lo eth0 END
got str=12381819273918723
           ajsdfkljaksdfjfkasdf
           eth0

Reply:
try this if garbage string happen
($iflist) =~ s/(.*lo\s)(.*)(\sEND.*)/$2/;

Reply:
Same as before.

str=[ 90.196160] dbus-run-session[7501]: dbus-daemon[7501]: Successfully activated service 'org.a11y.atspi.Registry'
[ 90.197403] dbus-run-session[7939]: SpiRegistry daemon is running with well-known name - org.a11y.atspi.Registry
lo eth0 END
got str=[ 90.196160] dbus-run-session[7501]: dbus-daemon[7501]: Successfully activated service 'org.a11y.atspi.Registry'
[ 90.197403] dbus-run-session[7939]: SpiRegistry daemon is running with well-known name - org.a11y.atspi.Registry
eth0

Reply:
@coolgw Use your new code, it works.   ($iflist) =~ s/(.*lo\s)(.*)(\sEND.*)/$2/s;

Wait for the verification log:
http://openqa.nue.suse.com/tests/5125727#live